ddtrace/contrib/trace_utils.py

```diff
@@ -577,7 +577,10 @@ def activate_distributed_headers(tracer, int_config=None, request_headers=None,
         #     app = Flask(__name__)  # Traced via Flask instrumentation
         #     app = DDWSGIMiddleware(app)  # Extra layer on top for WSGI
         current_context = tracer.current_trace_context()
-        if current_context and current_context.trace_id == context.trace_id:
+
+        # We accept incoming contexts with only baggage or only span_links, however if we
+        # already have a current_context then a baggage only or span_link only context will be tossed out
+        if current_context and (not context.trace_id or current_context.trace_id == context.trace_id):
```

Comment:
Thanks for the catch, fixing and updating comment

Reply:
However will continue to throw out an incoming context that lacks a trace_id as the actions around merging baggage and appending span links etc could lead to a mess and I think is basically never what someone would expect to happen.